dmprd.py

```diff
@@ -94,22 +97,26 @@ def __init__(self, multicast_address: str, port: int,
         addrinfo = socket.getaddrinfo(multicast_address, None)[0]
         iface_index = socket.if_nametoindex(interface_name)
 
-        super(MulticastRxSocket, self).__init__(addrinfo[0], socket.SOCK_DGRAM)
+        super(MulticastRxSocket, self).__init__(addrinfo[0], socket.SOCK_DGRAM, socket.IPPROTO_UDP)
 
         if hasattr(socket, 'SO_REUSEADDR'):
             self.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
 
         self.bind((multicast_address, port))
 
+        # bind the socket to the right interface
+        # no idea why its 25, just found it somewhere in the internet and it works
+        self.setsockopt(socket.SOL_SOCKET, 25, interface_name.encode('utf-8'))
         if addrinfo[0] == socket.AF_INET:
             ip_mreqn = get_ip_mreqn_struct(multicast_address, interface_address,
-                                           interface_name)
-            self.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP,
-                            ip_mreqn)
+                                          interface_name)
 
             # Allow looping if MCAST_LOOP is set to 1
-            self.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_LOOP,
-                            MCAST_LOOP)
+            self.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_LOOP, 0)
+            self.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, 8)
```

Comment:
This should be configurable on an interfaces basis. 8 hops is probably to high for many standard environments. The default should be 1 hop forwarding and only if the users knows the effects he should overwrite this value to a higher one in the configuration file.

This should also be specified in the DMPR protocol itself.

Reply:
removed ttl configuration for RX socket because its not needed there
in TX socket, it is already configurable
Changes verified with mininet emulation